File: sdk-node/examples/forms/remove-logic-rule.ts
/**
 * Remove the logic rule that reveals a target element from a source element.
 *
 * Run:
 *   npx tsx examples/forms/remove-logic-rule.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const element = await client.forms.removeLogicRule(
  'frm_replace_with_real_id',
  'elm_source_id',
  'elm_target_id',
)

console.log(`removed logic rule from element ${String(element.id)} (${element.type})`)

File: sdk-python/examples/forms/remove_logic_rule.py
"""Run with: python examples/forms/remove_logic_rule.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        element = client.forms.remove_logic_rule(
            "frm_replace_with_real_id",
            "elm_source_id",
            "elm_revealed_id",
        )
        print(f"removed logic rule from element {element.id} ({element.type})")


if __name__ == "__main__":
    main()
